fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/RankIndexTest.java

```diff
@@ -1116,8 +1116,7 @@ public void headerRankAndIdQuery() throws Exception {
                 .build();
         planner.setIndexScanPreference(QueryPlanner.IndexScanPreference.PREFER_SCAN);
         RecordQueryPlan plan = planner.plan(query);
-        assertEquals("Scan(([buffaloes, 100],[buffaloes]]) | [HeaderRankedRecord]" +
-                        " | score LESS_THAN $__rank_0 WHERE __rank_0 = score_by_nested_id.score_for_rank_else_skip(buffaloes, 2)",
+        assertEquals("Fetch(Covering(Index(score_by_nested_id ([buffaloes, null],[buffaloes, 2]) BY_RANK) -> [score: KEY[1], header: [group: KEY[0], id: KEY[2]]]) | header/{id GREATER_THAN 100})",
```

Comment:
This plan change is due to different _costing_. Previously `SCAN()` and `INDEX_SCAN()` where intrinsically considered equal and the `PREFER_SCAN` configuration parameter was the tie breaker. Now, the index is considered better as the filter on `header/{id GREATER THAN 100}` can be applied on the index key. Not sure if we want it like this.

Reply:
`SCAN` gives us something that is capable of evaluating the predicate with no further database traffic. More data is fetched in the scan itself. Whether this matters depends on the selectivity: if most pass, then we end up doing a fetch of the record anyway, using a less-efficient (even in parallel) single point fetch.

I kind of expected that _all_ filters to a `SCAN` would be index filters and not residuals. With that change, the same tie would result, restoring the status quo, right?

Reply:
It does! I have added that functionality.